src/onINP.ts

```diff
@@ -119,6 +119,26 @@ export const onINP = (onReport: ReportCallback, opts?: ReportOpts) => {
       if (entry.interactionId) {
         processEntry(entry);
       }
+
+      // Entries of type `first-input` don't currently have an `interactionId`,
+      // so to consider them in INP we have to first check that an existing
+      // entry doesn't match the `duration` and `startTime`.
+      // Note that this logic assumes that `event` entries are dispatched
+      // before `first-input` entries. This is true in Chrome but it is not
+      // true in Firefox; however, Firefox doesn't support interactionId, so
+      // it's not an issue at the moment.
```

Comment:
I was surprised to learn this, since I think we fire FID first.

However, I guess you mean that the order of Entries in the PO callbacks, when there are multiple observe() calls of different types... This I haven't tested and I'm not sure if order is guaranteed or arbitrary right now.

Reply:
Yeah, I only tested this in Chrome on Mac (desktop), but no matter what I did, which order I called `observe()` in, or which event I used, I always got the `event` entry processed before the `first-input` entry.